cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/sqlite/SQLDatabaseFactory.java

```diff
@@ -206,6 +206,8 @@ public static void updateSchema(SQLDatabase database, Migration migration, int v
                 }
             } finally {
                 database.endTransaction();
+                // re-enable foreign keys
+                database.execSQL("PRAGMA foreign_keys = ON;");
```

Comment:
Do we need to surround `endTransaction()` in a try/finally as well to ensure we turn the foreign keys back on? I worry if it could throw a RuntimeException even if it doesn't declare any other exceptions - although at that point we probably have bigger problems.